mtc.go

```diff
@@ -899,6 +924,80 @@ func (a *AbridgedAssertion) unmarshal(s *cryptobyte.String) error {
 	return nil
 }
 
+func (a *AssertionRequest) UnmarshalBinary(data []byte) error {
+	var (
+		s cryptobyte.String = cryptobyte.String(data)
+	)
+	a.Checksum = make([]byte, sha256.Size)
+	if !s.CopyBytes(a.Checksum) {
+		return ErrTruncated
+	}
+
+	err := a.Assertion.unmarshal(&s)
+	if err != nil {
+		return err
+	}
+
+	err = a.Evidence.unmarshal(&s)
+	if err != nil {
+		return err
+	}
+
+	if !s.Empty() {
+		return ErrExtraBytes
+	}
+
+	err = a.Check()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *AssertionRequest) marshalAndCheckAssertion() ([]byte, error) {
+	buf, err := a.Assertion.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	checksum2 := sha256.Sum256([]byte(buf))
```

Comment:
Wouldn't we want the checksum to also cover the evidence?

Reply:
I wasn't sure on this, but I think you're right. IIUC the purpose of the checksum is to catch bit flips in the queued assertions before we issue them, and we'd also want to catch bit flips in the evidence. Will update!